build.sh

```diff
@@ -1,48 +1,96 @@
+        command_options+=(
+            --extra-ldflags=-L${HOMEBREW_LOCATION}/lib
+            --extra-cflags=-I${HOMEBREW_LOCATION}/include
+        )
     fi
-    echo "configuring with command_options=${command_options}"
-    ${CONFIGURE_TO_USE} ${command_options}
+    echo "configuring with command_options=${command_options[@]}"
 elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
-    if [ "$1" == "--DEBUG" ] || [ "$1" == "--debug" ]; then
-        command_options="--prefix=${DIST} --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-shared --enable-avresample --enable-pthreads --enable-version3 --enable-hardcoded-tables --cc=clang --host-cflags=-fPIC --host-ldflags= --extra-cflags=-fPIC --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libx265 --enable-libxvid --enable-opencl --disable-lzma --disable-stripping --enable-debug=3 --extra-cflags=-ggdb --extra-cflags=-O0 --extra-cflags=-g"
-        CONFIGURE_TO_USE="./configure.debug"
-    else
-        command_options="--prefix=${DIST} --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-shared --enable-avresample --enable-pthreads --enable-version3 --enable-hardcoded-tables --cc=clang --host-cflags=-fPIC --host-ldflags= --extra-cflags=-fPIC --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libx265 --enable-libxvid --enable-opencl --disable-lzma"
+    command_options+=(
+        --extra-cflags=-fPIC
+    )
+    if find /usr/local/cuda* -name version.txt | read; then
+        if [ -d /usr/local/cuda/include -a -d /usr/local/cuda/lib64 ]; then
+            command_options+=(
+                --enable-cuda
+                --enable-cuvid
+                --enable-nvenc
+                --enable-nonfree
+                --enable-libnpp
+                --extra-cflags=-I/usr/local/cuda/include
+                --extra-ldflags=-L/usr/local/cuda/lib64
+            )
+        else
+            echo "CUDA found, but not linked to /usr/local/cuda, so skipping setup"
+        fi
     fi
-    FLAGS_NVIDIA=""
-    if [ -d /usr/local/cuda/ ]
-    then
-        FLAGS_NVIDIA="--enable-cuda --enable-cuvid --enable-nvenc --enable-nonfree --enable-libnpp --extra-cflags=-I/usr/local/cuda/include --extra-ldflags=-L/usr/local/cuda/lib64"
-    fi
-    echo "configuring with command_options=${FLAGS_NVIDIA} ${command_options}"
-    ${CONFIGURE_TO_USE} ${FLAGS_NVIDIA} ${command_options}
+    echo "configuring with command_options=${command_options[@]}"
+    
 elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
     echo "MINGW32 Not yet supported"
+    exit 1
 elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ]; then
     echo "MINGW64 Not yet supported"
+    exit 1
 fi
 
+(
```

Comment:
@elv-reza I added the execution in a sub-shell to make sure the correct directory was in use.  Reminded of this when I saw your question about changing `DIST`